core/src/main.cpp

```diff
@@ -495,11 +515,16 @@ auto print_configuration_template() -> void
         "zone": "<string>",
 
         "tls": {{
-            "client_server_policy": "<string>",
+            "client_server_policy": "CS_NEG_REFUSE",
             "ca_certificate_file": "<string>",
             "certificate_chain_file": "<string>",
             "dh_params_file": "<string>",
-            "verify_server": "<string>"
+            "verify_server": "cert",
+            "client_server_negotiation": "request_server_negotiation",
+            "encryption_algorithm": "AES-256-CBC",
+            "encryption_key_size": 32,
+            "encryption_hash_rounds": 16,
+            "encryption_salt_size": 8
```

Comment:
Are these required when the `client_server_policy` is `CS_NEG_REFUSE`?

Just wondering in light of the default/suggested values being used in combination here.

Reply:
they are not required if `CS_NEG_REFUSE` is used.

Reply:
@trel is correct.

Reply:
Well, the values of the properties are ignored by the iRODS libraries when CS_NEG_REFUSE is set.

All of the properties are required, but that only matters if the `tls` property exists in the config file.

Reply:
agreed.

Reply:
@alanking Does that answer your question?

Just to be even more clear, I'm guessing the client just sends whatever is defined. It's the iRODS server that decides whether it needs to worry about the other properties.

Reply:
Oh, yes, it does. I think I was curious about the options being there while `CS_NEG_REFUSE` is the value we provide by default. I think it's well understood and the folks who care will know what to do because they need to have SSL/TLS enabled on the iRODS server to use it anyways, like you guys said. Resolving